docs/sphinx/releases.rst

```diff
@@ -4,10 +4,21 @@ CUDA Quantum Releases
 
 **latest**
 
-The latest version of CUDA Quantum is on the main branch of our `GitHub repository <https://github.com/NVIDIA/cuda-quantum>`__.
+The latest version of CUDA Quantum is on the main branch of our `GitHub repository <https://github.com/NVIDIA/cuda-quantum>`__, as well as available as Docker image. More information about installing the nightly builds can be found :doc:`here <install>`
 
+- `Docker image (nightly builds) <https://catalog.ngc.nvidia.com/orgs/nvidia/teams/nightly/containers/cuda-quantum>`__
 - `Documentation <https://nvidia.github.io/cuda-quantum/latest>`__
 
+**0.4.1**
+
+The 0.4.1 release adds support for ARM processors in the form of multi-platform Docker images and `aarch64` Python wheels. Additionally, all GPU-based backends are now included in the Python wheels as well as in the Docker image.
+
+- `Docker image <https://catalog.ngc.nvidia.com/orgs/nvidia/containers/cuda-quantum>`__
+- `Python wheel <https://pypi.org/project/cuda-quantum/>`__
+- `Documentation <https://nvidia.github.io/cuda-quantum/0.4.1>`__
```

Comment:
You probably already know this, but just to be sure, this is currently a broken link. The only way around this problem (that I see) is first placing this commit on the `releases/*` branch and then cherry-pick this commit back to `main` once the release is complete. I assume we don't want to do that?

Reply:
Actually, I didn't know this - and I need to set up a link check for rst files...
What you outline as a way around this is not necessary. 

Here is how it should work:
When a release branch is created, the deployment workflow is triggered that produces documentation for that version as an artifact. The successful completion of that workflow in turn triggers two other workflows one for the publishing of the docker images (as nightly builds) and one for the publishing of docs. Hence, as soon as the release branch exists (or at least an hour or so after its creation), so should the docs for that version (i.e. the link above). 
Regarding the fact that releases.rst is updated only on main, this is in fact sufficient for it to be show in versions of the docs (otherwise we would need to go update this file on all previous release branches...). This is achieved by the redirect of <whatever docs version>/versions.rst to latest/releases.rst.

What went wrong:
I misconfigured the deployment that happens when a release branch is created. If you look at [that deployment](https://github.com/NVIDIA/cuda-quantum/actions/runs/6323804387), what is wrong with it is that it only runs the full deployment for the multi-platform build and is triggered by the user who created the release branch. This has two consequences I need to fix: 
1) It does not create separate build caches for each architecture, which is needed for the caches to work properly.
2) The two publishing workflows are only triggered automatically when the deployment was also triggered automatically, ie. if the triggering actor is the cuda-quantum-bot.

Reply:
Here are the necessary changes: https://github.com/NVIDIA/cuda-quantum/pull/709